src/web/components/App.tsx

```diff
@@ -187,14 +187,19 @@ export const App = ({ CAPI, NAV }: Props) => {
         const callGetUser = async () => {
             setUser(await getUser(CAPI.config.discussionApiUrl));
         };
-        const callGetBrazeUuid = async () => {
-            setAsyncBrazeUuid(getBrazeUuid(CAPI.config.idApiUrl));
-        };
+
         if (isSignedIn) {
             callGetUser();
-            callGetBrazeUuid();
         }
-    }, [isSignedIn, CAPI.config.discussionApiUrl, CAPI.config.idApiUrl]);
+    }, [isSignedIn, CAPI.config.discussionApiUrl]);
+
+    useEffect(() => {
+        if (isSignedIn) {
+            setAsyncBrazeUuid(getBrazeUuid(CAPI.config.idApiUrl));
+        } else {
+            setAsyncBrazeUuid(Promise.resolve(null));
```

Comment:
could you instead pass `Promise.resolve(null)` as the initial value when you call `useState`? Would the behaviour be the same?

Reply:
Annoyingly I don't think that quite works in the case where the user is signed in and there _is_ a braze UUID. Since we'll use the first promise and the banner logic will only run once (by design, since we only want to show one banner per page view).

Now we'll wait until the `asyncBrazeUuid` is defined (which should always happen now) before running the banner picker.